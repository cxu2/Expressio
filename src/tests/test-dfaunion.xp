main : -> int {
int ans;
dfa zzo;
dfa oo;
dfa u;
zzo  = {
 states : 4
 alphabet : ['0', '1']
 start : 0
 final : [3]
 transitions : [ (0 '0' 1), (1 '0' 2), (2 '1' 3) ] 
};

oo = {
 states : 3
 alphabet : ['0', '1']
 start : 0
 final : [2]
 transitions : [ (0 '1' 1), (1 '1' 2) ]
};


u = zzo union oo;
printdfa(u);
  return 0;
}
