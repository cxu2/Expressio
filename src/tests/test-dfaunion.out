u rejects 00
u rejects 11
u rejects 0
u rejects 111
